projects/plugins/jetpack/modules/subscriptions/subscribe-modal/class-jetpack-subscribe-modal.php

```diff
@@ -0,0 +1,261 @@
+					'accept-charset'         => array(),
+					'data-blog'              => array(),
+					'data-post_access_level' => array(),
+					'id'                     => array(),
+				),
+				'label'  => array(
+					'id'    => array(),
+					'for'   => array(),
+					'class' => array(),
+				),
+				'input'  => array(
+					'id'          => array(),
+					'type'        => array(),
+					'name'        => array(),
+					'value'       => array(),
+					'class'       => array(),
+					'style'       => array(),
+					'placeholder' => array(),
+					'required'    => array(),
+				),
+				'button' => array(
+					'type'  => array(),
+					'class' => array(),
+					'style' => array(),
+					'name'  => array(),
+				),
+			);
+
+			?>
+				<div class="jetpack-subscribe-modal">
+					<div class="jetpack-subscribe-modal__modal-content">
+						<?php
+						foreach ( $blocks as $block ) {
+							echo wp_kses( render_block( $block ), $allowed_html );
+						}
+						?>
+						<Dashicon icon="minus" className="jp-idc__idc-screen__card-action-separator" />
+					</div>
+				</div>
+			<?php
+		}
+	}
+
+	/**
+	 * Enqueues JS to load modal.
+	 *
+	 * @return void
+	 */
+	public function wpcom_enqueue_subscribe_modal_assets() {
+		if ( $this->is_front_end_single_post() ) {
+			wp_enqueue_style( 'subscribe-modal-css', plugins_url( 'subscribe-modal.css', __FILE__ ), array(), JETPACK__VERSION );
+			wp_enqueue_script( 'subscribe-modal-js', plugins_url( 'subscribe-modal.js', __FILE__ ), array( 'wp-components' ), JETPACK__VERSION, true );
+		}
+	}
+
+	/**
+	 * Returns true if we should enable Newsletter content.
+	 * This is currently limited to lettre theme or newsletter sites.
+	 * We could open it to all themes or site intents.
+	 *
+	 * @return bool
+	 */
+	public function should_enable_subscriber_modal() {
+		$is_lettre          = get_option( 'stylesheet' ) === 'lettre';
+		$is_newsletter_site = get_option( 'site_intent' ) === 'newsletter';
+		$is_modal_enabled   = get_option( 'sm_enabled' ) || false;
+		$is_fse_theme       = wp_is_block_theme();
+
+		$should_enable_subscriber_modal = ( $is_lettre || $is_newsletter_site ) && $is_fse_theme && $is_modal_enabled;
+
+		/**
+		* Allows force-enabling or disabling the Subscriptions modal.
+		*
+		* @since $$next-version$$
+		*
+		* @param bool $should_enable_subscriber_modal Whether the Subscriptions modal should be enabled.
+		*/
+		return (bool) apply_filters( 'jetpack_subscriptions_modal_force_enabled', $should_enable_subscriber_modal );
+	}
+
+	/**
+	 * Returns true if we we are on frontend of single post.
+	 * Note: Because of how WordPress works, this function will
+	 * only return the correct value after a certain point in the
+	 * WordPress loading process. You cannot, for example, use this
+	 * method in the class contructor method - it will always return false.
+	 *
+	 * @return bool
+	 */
+	public function is_front_end_single_post() {
+		return ! is_admin() && is_single() && 'post' === get_post_type();
```

Comment:
You should be able to use `is_singular` here:

```suggestion
		return ! is_admin() && is_singular( 'post' );
```